src/Nancy/IO/RequestStream.cs

```diff
@@ -358,9 +358,23 @@ private void ShiftStreamToFileStreamIfNecessary()
 
         private static FileStream CreateTemporaryFileStream()
         {
-            var filePath = Path.GetTempFileName();
-
-            return new FileStream(filePath, FileMode.Create, FileAccess.ReadWrite, FileShare.None, 8192, StaticConfiguration.AllowFileStreamUploadAsync);
+            try
+            {
+                var filePath = Path.GetTempFileName();
+
+                return new FileStream(
+                    filePath,
+                    FileMode.Create,
+                    FileAccess.ReadWrite,
+                    FileShare.None,
+                    8192,
+                    StaticConfiguration.AllowFileStreamUploadAsync);
+            }
+            catch (IOException ex)

```

Comment:
I'd have to declare a FileStream variable outside the try{} block and then return it after, this is, imo, cleaner.

Even better would be 
```
return new FileStream(Path.GetTempFileName(), FileMode.Create, FileAccess.ReadWrite, FileShare.None, 8192, StaticConfiguration.AllowFileStreamUploadAsync);
```